driver-core/src/test/java/com/datastax/driver/core/querybuilder/QueryBuilderTest.java

```diff
@@ -999,4 +994,63 @@ public void should_include_original_cause_when_arguments_invalid() {
         }
     }
 
+    static class Foo {
+        int bar;
+
+        public Foo(int bar) {
+            this.bar = bar;
+        }
+    }
+
+    static class FooCodec extends TypeCodec<Foo> {
+
+        public FooCodec() {
+            super(DataType.cint(), Foo.class);
+        }
+
+        @Override
+        public ByteBuffer serialize(Foo value, ProtocolVersion protocolVersion) throws InvalidTypeException {
+            // not relevant for this test
+            return null;
+        }
+
+        @Override
+        public Foo deserialize(ByteBuffer bytes, ProtocolVersion protocolVersion) throws InvalidTypeException {
+            // not relevant for this test
+            return null;
+        }
+
+        @Override
+        public Foo parse(String value) throws InvalidTypeException {
+            // not relevant for this test
+            return null;
+        }
+
+        @Override
+        public String format(Foo foo) throws InvalidTypeException {
+            return Integer.toString(foo.bar);
+        }
+    }
+
+    /**
+     * Ensures that a statement can be printed with and without
+     * a required custom codec.
+     * The expectation is that if the codec is not registered,
+     * then the query string should contain bind markers for all variables;
+     * if however all codecs are properly registered, then
+     * the query string should contain all variables inlined and formatted properly.
+     *
+     * @jira_ticket JAVA-1272
+     */
+    @Test(groups = "unit")
+    public void should_inline_custom_codec() throws Exception {
+        assertThat(
+                insertInto("users").value("id", new Foo(42)).toString())
+                .isEqualTo("INSERT INTO users (id) VALUES (?);");
+        CodecRegistry.DEFAULT_INSTANCE.register(new FooCodec());
```

Comment:
Since `DEFAULT_INSTANCE` is static, this will impact all other tests, but since Foo is only used here I don't see it being a problem.